src/nested_pandas/nestedframe/io.py

```diff
@@ -246,17 +256,18 @@ def _read_remote_parquet_directory(
     """Read files one-by-one with fsspec.open_parquet_file and concat the result"""
     tables = []
     for upath in dir_upath.iterdir():
-        with fsspec.parquet.open_parquet_file(
-            upath.path,
-            columns=columns,
-            storage_options=storage_options,
-            fs=filesystem,
-            engine="pyarrow",
-        ) as parquet_file:
-            # Go recursively for supported filesystems, etc. for S3, but not for HTTP(S).
-            if _is_fh_a_dir(parquet_file):
-                table = _read_remote_parquet_directory(upath, filesystem, storage_options, columns, **kwargs)
-            else:
+        # Go recursively for filesystems which support file/directory identification with fsspec file
+        # handlers. This would work for e.g. S3, but not for HTTP(S).
+        if _is_remote_dir(upath, upath):
```

Comment:
If `UPath` does remove trailing `/`, then doesn't this prevent `_is_remote_dir` from ever going recursive? Or does `.iterdir()` return the trailing slash for subdirectories it finds?